packages/adapters/bitcoin/src/adapter.ts

```diff
@@ -56,33 +56,39 @@ export class BitcoinAdapter extends AdapterBlueprint<BitcoinConnector> {
       throw new Error('The connector does not support any of the requested chains')
     }
 
-    const connection = this.getConnection({
-      address: params.address,
-      connectorId: connector.id,
-      connections: this.connections,
-      connectors: this.connectors
-    })
-
-    if (connection?.account) {
-      this.emit('accountChanged', {
-        address: connection.account.address,
-        chainId: connection.caipNetwork?.id,
-        connector
+    // If we are already connected to the address, return the connection to prevent extra interactions
+    if (params.address) {
```

Comment:
Should we do for the other adapters as well? i.e Solana or Ton also have the same logics, or it's Bitcoin only issue?